packages/hierarchy-builder/src/hierarchy-builder/internal/operators/grouping/GroupHiding.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { HierarchyNode } from "../../../HierarchyNode";
+import { GroupingHandlerReturn } from "../Grouping";
+
+/**
+ * @internal
+ */
+export function applyGroupHidingParams(allNodes: HierarchyNode[]): GroupingHandlerReturn {
+  const finalGroupings: GroupingHandlerReturn = { allNodes: [], groupedNodes: [] };
+  for (const node of allNodes) {
+    if (HierarchyNode.isGroupingNode(node)) {
+      if (Array.isArray(node.children) && (allNodes.length === 1 || node.children.length === 1)) {
+        const [hideIfNoSiblings, hideIfOneGroupedNode] = getGroupingHideOptionsFromParentNode(node);
+        if (hideIfNoSiblings && allNodes.length === 1) {
+          return { allNodes: node.children, groupedNodes: [] };
+        } else if (hideIfOneGroupedNode && node.children.length === 1) {
+          finalGroupings.allNodes.push(node.children[0]);
+          continue;
+        }
+      }
+      finalGroupings.groupedNodes.push(node);
+    }
+    finalGroupings.allNodes.push(node);
+  }
+  return finalGroupings;
+}
+
+/**
+ * @internal
+ */
+function getGroupingHideOptionsFromParentNode(parentNode: HierarchyNode): [hideIfNoSiblings: boolean, hideIfOneGroupedNode: boolean] {
+  if (Array.isArray(parentNode.children)) {
+    if (HierarchyNode.isBaseClassGroupingNode(parentNode)) {
+      return getHideOptionsFromBaseClassGroupingNodes(parentNode.children);
+    }
+    if (HierarchyNode.isClassGroupingNode(parentNode)) {
+      return getHideOptionsFromClassGroupingNodes(parentNode.children);
+    }
+    if (HierarchyNode.isLabelGroupingNode(parentNode)) {
+      return getHideOptionsFromLabelGroupingNodes(parentNode.children);
+    }
```

Comment:
I don't think we need 3 different functions for this. Maybe something like this would work?
```ts
function getHideOptionsFromNodeProcessingParams(nodes: HierarchyNode[], hideOptionsAccessor: (params: HierarchyNodeHandlingParams) => BaseGroupingParams | undefined): [hideIfNoSiblings: boolean, hideIfOneGroupedNode: boolean] {
  let hideIfNoSiblings = false;
  let hideIfOneGroupedNode = false;
  for (const node of nodes) {
    if (hideIfNoSiblings && hideIfOneGroupedNode) {
      break;
    }
    const params = node.params ? hideOptionsAccessor(node.params) : undefined;
    hideIfNoSiblings = !!params.hideIfNoSiblings;
    hideIfNoSiblings = !!params.hideIfOneGroupedNode;
  }
  return [hideIfNoSiblings, hideIfOneGroupedNode];
}

// usage:
if (HierarchyNode.isBaseClassGroupingNode(parentNode)) {
  return getHideOptionsFromNodeProcessingParams(parentNode.children, (p) => p.grouping?.byBaseClasses);
}
if (HierarchyNode.isClassGroupingNode(parentNode)) {
  return getHideOptionsFromNodeProcessingParams(parentNode.children, (p) => typeof p.grouping?.byClass === "object" ? p.grouping.byClass : undefined);
}
if (HierarchyNode.isLabelGroupingNode(parentNode)) {
  return getHideOptionsFromNodeProcessingParams(parentNode.children, (p) => typeof p.grouping?.byLabel === "object" ? p.grouping.byLabel : undefined);
}
```

Reply:
Changed to your suggestion, also, changed this suggestion:
```
hideIfNoSiblings = !!params.hideIfNoSiblings;
hideIfNoSiblings = !!params.hideIfOneGroupedNode;
```
to this:
```
hideIfNoSiblings ||= !!params.hideIfNoSiblings;
hideIfOneGroupedNode ||= !!params.hideIfOneGroupedNode;
```
Since only one node needs to have these options checked and not all of them.